src/main/java/com/coveo/spillway/storage/AsyncBatchLimitUsageStorage.java

```diff
@@ -110,4 +136,27 @@ public AsyncBatchLimitUsageStorage(
   public void close() throws Exception {
     wrappedLimitUsageStorage.close();
   }
+
+  /**
+   * When a request is made, we want to make sure the cache has the latest value from storage before enforcing.
+   * To do so, we issue a zero-cost request for limits that are not cached already and we trigger a sync to retrieve
+   * the actual value from the storage to the cache.
+   */
+  private void initMissingLimitsFromInMemoryCache(Collection<AddAndGetRequest> requests) {
+    Set<LimitKey> currentLimitCountersKeys = cache.getCurrentLimitCounters().keySet();
+    List<AddAndGetRequest> missingLimitsRequests =
+        requests
+            .stream()
+            .filter(request -> !currentLimitCountersKeys.contains(LimitKey.fromRequest(request)))
+            .collect(Collectors.toList());
+    if (missingLimitsRequests.size() > 0) {
+      cache.addAndGet(
```

Comment:
This looks a bit like a hack - the clear meaning of what you want to do here is to retrieve the initial values for the missing keys. Couldn't you add an explicit method to `CacheSynchronization` for this?

Reply:
No, I have tried this and the cache synchronization works only for known keys in the inmemory cache. It does not retrieve all keys from remote storage.

Although it looks like a hack, issuing a zero cost request is also done to "try" a call from the main interface.

To do what you are suggesting, I would have to get all values from remote storage at init time. That way, I could clear the request path. The remaining flakyness would be near the thresholds as it is expected I believe. I will give it a try.